### PR TITLE
[#25] [#26] [API] As a user, I can get the list of keywords, [API] As a user, I can filter the list of keywords

### LINKED ISSUE #25: [API] As a user, I can filter the list of keywords
## Why

Create an API for search across all reports

## Acceptance Criteria

Create a new route and endpoint

### LINKED ISSUE #26: [API] As a user, I can get the list of keywords
## Why

Users can get their uploaded keywords.

## Acceptance Criteria

- Create a new route and endpoint.
- Contain metadata for pagination.

### PR DESCRIPTION
**Pull request by vnntsu**
close #25
close #26

## What happened 👀

- Create API to get list of keyword `{GET /api/v1/keyword}`
- It includes filtering by: `{GET /api/v1/keyword/?keyword={text}}`

## Insight 📝

- Create a new route
- Add unit test

## Proof Of Work 📹

Screenshot 2023-02-06 at 18 24 06

### CODE REVIEW

File: test/crawler_web/controllers/api/v1/keyword_controller_test.exs [hunk 1]
```diff
@@ -3,7 +3,101 @@ defmodule CrawlerWeb.Api.V1.KeywordControllerTest do
 
   alias Crawler.Keyword.Keywords
 
-  describe "POST /keyword/upload" do
+  describe "GET index/2" do
+    test "given a valid user, returns uploaded keywords", %{conn: conn} do
```
Comment (andyduong1920) [inline test/crawler_web/controllers/api/v1/keyword_controller_test.exs:7]: should we test for `given an unauthenticated user`

File: test/crawler_web/controllers/api/v1/keyword_controller_test.exs [hunk 2]
```diff
@@ -3,7 +3,101 @@ defmodule CrawlerWeb.Api.V1.KeywordControllerTest do
...
+      user = insert(:user)
+      insert(:keyword, name: "watch", user_id: user.id)
```
Comment (andyduong1920) [inline test/crawler_web/controllers/api/v1/keyword_controller_test.exs:9]: same the other PR, let add some keywords that has the same `name` but that doesn't belongs into the logged in user, so we could make sure this controller only return the user's keywords, what do you think?

File: test/crawler_web/controllers/api/v1/keyword_controller_test.exs (hunk 1 above)
Comment (vnntsu) [reply test/crawler_web/controllers/api/v1/keyword_controller_test.exs:7]: Added in [f8993ba](https://github.com/vnntsu/elixir-ic/pull/46/commits/f8993ba360e5baffcf585fe5ac362bcfbb3f53e6)

File: test/crawler_web/controllers/api/v1/keyword_controller_test.exs (hunk 2 above)
Comment (vnntsu) [reply test/crawler_web/controllers/api/v1/keyword_controller_test.exs:9]: Fixed in [f8993ba](https://github.com/vnntsu/elixir-ic/pull/46/commits/f8993ba360e5baffcf585fe5ac362bcfbb3f53e6)

**Review by andyduong1920 — APPROVED** [review]

**Review by Nihisil — APPROVED** [review]

**Pull request merged**